src/lib.rs

```diff
@@ -12,127 +12,199 @@
-//! 2. [IOread](trait.IOread.html), for reading _simple_ data out of a `std::io::Read` based interface, e.g., a stream. (**Note**: only available when compiled with `std`)
+//! Scroll sets down a number of traits:
 //!
-//! Each of these interfaces also have their corresponding writer versions as well, i.e. [Pwrite](trait.Pwrite.html) and [IOwrite](trait.IOwrite.html), respectively.
+//! [FromCtx](ctx/trait.FromCtx.html), [IntoCtx](ctx/trait.IntoCtx.html) and
+//! [TryFromCtx](ctx/trait.TryFromCtx.html); to be implemented on custom types to allow reading,
+//! writing, and potentially fallible reading respectively.
 //!
-//! Most familiar will likely be the `pread` method from the `Pread` trait (inspired from the C function), which in our case takes an immutable reference to self, an immutable offset to read at, (and _optionally_ a parsing context, more on that later), and then returns the deserialized value.
+//! [Pread](trait.Pread.html) and [Pwrite](trait.Pwrite.html) which are implemented on data
+//! containers such as byte arrays to define how to read or respectively write types implementing
+//! the *Ctx traits above.
+//! In addition scroll also defines [IOread](trait.IOread.html) and
+//! [IOwrite](trait.IOwrite.html) with additional constraits that then allow reading and writing
+//! from `std::io` [Read](https://doc.rust-lang.org/nightly/std/io/trait.Read.html) and
+//! [Write](https://doc.rust-lang.org/nightly/std/io/trait.Write.html).
 //!
-//! Because self is immutable, _**all** reads can be performed in parallel_ and hence are trivially parallelizable.
 //!
-//! For most usecases, you can use [scroll_derive](https://docs.rs/scroll_derive) to annotate your types with `derive(Pread, Pwrite, IOread, IOwrite, SizeWith)` to automatically add sensible derive defaults, and you should be ready to roll.  For more complex usescases, you can implement the conversion traits yourself, see the [context module](ctx/index.html) for more information.
+//! In most cases you can use [scroll_derive](https://docs.rs/scroll_derive) to derive sensible
+//! defaults for `Pread`, `Pwrite`, their IO counterpart and `SizeWith`.  More complex situations
+//! call for manual implementation of those traits; refer to [the ctx module](ctx/index.html) for
+//! details.
 //!
-//! # Example
 //!
-//! A simple example demonstrates its flexibility:
+//! ## Efficiently:
+//!
+//! Reading Slices — including [&str](https://doc.rust-lang.org/std/primitive.str.html) — supports
+//! zero-copy. Scroll is designed with a `no_std` context in mind; every dependency on `std` is
+//! cfg-gated and errors need not allocate.
+//!
+//! Reads by default take only immutable references wherever possible, allowing for trivial
+//! parallelization.
+//!
+//! # Examples
+//!
+//! Let's start with a simple example
 //!
 //! ```rust
-//! use scroll::{ctx, Pread, LE};
+//! use scroll::{ctx, Pread};
+//!
+//! // Let's first define some data, cfg-gated so our assertions later on hold.
```

Comment:
Notably, I have not actually checked if these examples work on big endian architectures. ^^'